### ISSUE TITLE
Use get only should not warn here

### ISSUE THREAD

**Issue by JohanLarsson** [GuOrg/Gu.Analyzers#135]
```cs
namespace RoslynSandbox
{
    public class Foo<T>
        where T : struct 
    {
        public T Value { get; private set; }

        public T? Meh
        {
            set { Value = value.HasValue ? value.Value : default(T); }
        }
    }
}
```

Not sure it is a repro.
